main.tf

```diff
@@ -4,12 +4,14 @@ locals {
 
   db_rules = "${null_resource.subnet_mappings.*.triggers}"
 
-  vaultName = "infra-vault-${var.subscription}"
+  vaultName = var.key_vault_name != null ? var.key_vault_name : "infra-vault-${var.subscription}"
```

Comment:
don't you want to check empty string ""? or does that get converted to null

Reply:
Nope you are right I need "" as I've set the default to ""